### PR TITLE
Kanban card creation revamp

### LINKED ISSUE #6957: Set a record name on Kanban card creation
# Current behavior

When creating a new record on a kanban board, you cannot set a name for the record from the kanban view. To name the record, you must go to the record detail page.

This doesn't apply to the opportunities kanbans where a specific behavior applies.

https://github.com/user-attachments/assets/cbd69ebb-ec2e-404f-a171-30e769daa9db

# Desired Behavior

We should implement the same behavior we have on table views: when creating a new record, we should display an input to set the record name from the board view. If the input is left empty, the card should be deleted, and the record should not be saved.

This doesn't apply to Opportunities kanbans.

CleanShot 2024-09-09 at 17 34 55

The input label should be the object identifier. We should display the double input for people kanbans.

Table behavior to copy:

https://github.com/user-attachments/assets/8f58fdbc-fe75-44fc-a303-918f7fe78338

### PR DESCRIPTION
**Pull request by ehconitin**
fixes #6957

### CODE REVIEW

**Comment by ehconitin** [comment]
@Bonapara work is still in progress but let me know your thoughts from product perspective.
Thanks

https://github.com/user-attachments/assets/40c694e7-96de-452b-bccf-c2f2adb51330

**Comment by Bonapara** [comment]
Hi @ehconitin, thanks! Could you use the floating input instead of placing the input within the card?

image

**Comment by ehconitin** [comment]
@Bonapara will do!
@Bonapara Question - 
we do not have to set name for opportunity kanban right? it will still take the company/relations name? 
@lucasbordeau could you let me know if this approach is correct?

https://github.com/user-attachments/assets/aa6ab46d-b210-471c-bd29-47e9e463f582

**Comment by ehconitin** [comment]
https://github.com/user-attachments/assets/c71bee78-16b9-48ed-bd35-c85f78d97bb8

**Comment by Bonapara** [comment]
> Hi @ehconitin, thanks! Could you use the floating input instead of placing the input within the card?
> 
> image

Great! Can you apply this design? I was also wondering if we can make the `stage` appear on the card before the title is set and the record saved.

**Comment by ehconitin** [comment]
@Bonapara 
could u reshare the image, it seems to be broken on my side. refresh isnt helping

**Comment by ehconitin** [comment]
> > Hi @ehconitin, thanks! Could you use the floating input instead of placing the input within the card?
> > image
> 
> Great! Can you apply this design? I was also wondering if we can make the `stage` appear on the card before the title is set and the record saved.

Yup, on it!

**Comment by ehconitin** [comment]
@Bonapara 
I think having stage appear before creating a card will have many challenges, but let me ask @lucasbordeau if he could think of an easier way!
Floating input is done btw!
@lucasbordeau 
Could u also help me with hotscope? Its not working here in RecordBoardCard? enter or esc doesnt work. In recording I am clicking outside to create the card.

https://github.com/user-attachments/assets/ee4d69c0-caf7-4ca7-86d0-1a6ed20136bd

**Comment by ehconitin** [comment]
Should I address #7002 in this PR in a similar manner, or handle it separately? Let me know what works best!

File: packages/twenty-front/src/modules/ui/input/components/TextInput.tsx [hunk 1]
```diff
@@ -89,7 +89,6 @@ export const TextInput = ({
       onInputEnter?.();
 
       if (isDefined(inputRef) && 'current' in inputRef) {
-        inputRef.current?.blur();
```
Comment (ehconitin) [inline packages/twenty-front/src/modules/ui/input/components/TextInput.tsx:92]: why is it here???
doesnt this cause an unnecessary onblur call?

**Comment by ehconitin** [comment]
Label identifier issue fixed!

https://github.com/user-attachments/assets/93d8543d-5b79-4de6-9499-16d37ab3cdc5

**Comment by ehconitin** [comment]
@Bonapara there was a floating input already but styles like box shadow were missing, fixed it!

File: packages/twenty-front/src/modules/object-record/record-index/components/RecordIndexPageKanbanAddButton.tsx [hunk 2]
```diff
@@ -50,35 +58,30 @@ export const RecordIndexPageKanbanAddButton = () => {
...
-    [
-      isOpportunity,
-      createRecordWithoutCompany,
-      setHotkeyScopeAndMemorizePreviousScope,
-      closeDropdown,
-    ],
-  );
+  const { handleAddNewCardClick } = useAddNewCard();
+
+  const handleItemClick = (columnDefinition: RecordBoardColumnDefinition) => {
```
[the remark this answers is not in the document]
Comment (ehconitin) [reply packages/twenty-front/src/modules/object-record/record-index/components/RecordIndexPageKanbanAddButton.tsx]: I was trying something without the callback and forgot to reverse it :(
Thanks for pointing that out! mb

**Comment by ehconitin** [comment]
@Bonapara 
#7163 

> All inline field heights should be 24px, as they are on record pages.

> There should be a 2px gap between inline field rows.

image

File: packages/twenty-front/src/modules/ui/input/components/TextInput.tsx (hunk 1 above)
Comment (ehconitin) [reply packages/twenty-front/src/modules/ui/input/components/TextInput.tsx:92]: But if not removed, two cards are being created on Enter.
Which means when onEnter is called onBlur is triggered too. 
How do we handle this? or should we just not create a card onBlur and keep this as it is?
Which means card would be created only onEnter. :(

Comment (bosiraphael) [reply packages/twenty-front/src/modules/ui/input/components/TextInput.tsx:92]: Indeed

File: packages/twenty-front/src/modules/object-record/record-board/record-board-column/components/RecordBoardColumnNewButton.tsx [hunk 3]
```diff
@@ -21,13 +24,53 @@ const StyledButton = styled.button`
...
+  if (newRecord.isCreating && newRecord.position === 'last') {
+    return (
+      <RecordBoardCard
+        isCreating={true}
+        onCreateSuccess={() => handleCreateSuccess('last')}
+        position="last"
+      />
+    );
+  }
+
```
Comment (bosiraphael) [inline packages/twenty-front/src/modules/object-record/record-board/record-board-column/components/RecordBoardColumnNewButton.tsx:69]: Seems like duplicate code with `packages/twenty-front/src/modules/object-record/record-board/record-board-column/components/RecordBoardColumnHeader.tsx`

Comment (ehconitin) [reply packages/twenty-front/src/modules/object-record/record-board/record-board-column/components/RecordBoardColumnNewButton.tsx]: @bosiraphael Done! I had to create a new hook due to a scoping issue, as it couldn't be kept within `useAddNewCard`

**Review by bosiraphael — APPROVED** [review]
LGTM :)

**Pull request merged**